secret/secret_test.go

```diff
@@ -6,6 +6,7 @@ import (
 
```

Comment:
We should probably test that `UpdateSecrets()` clears the dirty flag (and that it's never actually written to the secret store).